### ISSUE TITLE
Installation error related to a missing font

### ISSUE THREAD

**Issue by jsbyrnes** [GenericMappingTools/pygmt#3042]
### Description of the problem

I am attempting to install pygmt on macOS Monterey (12.5.1), on a system for which I've been using pygmt for a long time. I am not sure what caused this bug to start, but I've been attempting to reinstall and consistently getting this error.

After following the installation, I enter the test code, and the error triggers on fig.show()

### Minimal Complete Verifiable Example

```python
import pygmt
pygmt.show_versions()

fig = pygmt.Figure()
fig.coast(region="g", frame=True, shorelines=1)
fig.show()
```

### Full error message

```python-traceback
*** Warning: GenericResourceDir doesn't point to a valid resource directory.
               the -sGenericResourceDir=... option can be used to set this.

Error: /invalidfont in findfont
Operand stack:   --nostringval--   Helvetica   Helvetica
Execution stack:   %interp_exit   .runexec2   --nostringval--   .findfontop   --nostringval--   2   %stopped_push   --nostringval--   .findfontop   .findfontop   false   1   %stopped_push   1928   1   3   %oparray_pop   1927   1   3   %oparray_pop   1912   1   3   %oparray_pop   1785   1   3   %oparray_pop   --nostringval--   %errorexec_pop   .runexec2   --nostringval--   .findfontop   --nostringval--   2   %stopped_push   --nostringval--   .findfontop   .findfontop   1878   3   4   %oparray_pop
Dictionary stack:   --dict:750/1123(ro)(G)--   --dict:0/20(G)--   --dict:85/200(L)--   --dict:157/250(L)--
Current allocation mode is local
Last OS error: No such file or directoryCurrent file position is 22219
psconvert [ERROR]: System call [gs -q -dNOSAFER -dNOPAUSE -dBATCH -sDEVICE=bbox -DPSL_no_pagefill -dMaxBitmap=2147483647 -dUseFastColor=true '/Users/josephbyrnes/.gmt/sessions/gmt_session.14993/gmt_1.ps-' 2> '/Users/josephbyrnes/.gmt/sessions/gmt_session.14993/psconvert_14993c.bb'] returned error 256.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/figure.py", line 468, in show
    pdf = self._preview(
          ^^^^^^^^^^^^^^
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/figure.py", line 495, in _preview
    self.savefig(fname, dpi=dpi, **kwargs)
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/figure.py", line 374, in savefig
    self.psconvert(prefix=prefix, fmt=fmt, crop=crop, **kwargs)
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/helpers/decorators.py", line 603, in new_module
    return module_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/helpers/decorators.py", line 776, in new_module
    return module_func(*bound.args, **bound.kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/figure.py", line 252, in psconvert
    lib.call_module(
  File "/Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/python3.12/site-packages/pygmt/clib/session.py", line 624, in call_module
    raise GMTCLibError(
pygmt.exceptions.GMTCLibError: Module 'psconvert' failed with status code 79:
psconvert [ERROR]: System call [gs -q -dNOSAFER -dNOPAUSE -dBATCH -sDEVICE=bbox -DPSL_no_pagefill -dMaxBitmap=2147483647 -dUseFastColor=true '/Users/josephbyrnes/.gmt/sessions/gmt_session.14993/gmt_1.ps-' 2> '/Users/josephbyrnes/.gmt/sessions/gmt_session.14993/psconvert_14993c.bb'] returned error 256.
```

### System information

```bash
PyGMT information:
  version: v0.11.0
System information:
  python: 3.12.1 | packaged by conda-forge | (main, Dec 23 2023, 08:05:03) [Clang 16.0.6 ]
  executable: /Users/josephbyrnes/opt/anaconda3/envs/pygmt/bin/python
  machine: macOS-12.5.1-x86_64-i386-64bit
Dependency information:
  numpy: 1.26.3
  pandas: 2.2.0
  xarray: 2024.1.1
  netCDF4: 1.6.5
  packaging: 23.2
  contextily: None
  geopandas: None
  ipython: None
  rioxarray: None
  ghostscript: 10.02.1
GMT library information:
  binary version: 6.5.0
  cores: 8
  grid layout: rows
  image layout: 
  library path: /Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/libgmt.dylib
  padding: 2
  plugin dir: /Users/josephbyrnes/opt/anaconda3/envs/pygmt/lib/gmt/plugins
  share dir: /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/gmt
  version: 6.5.0
```

**Comment by seisman** [comment]
What's your output for the `which gs` command?

**Comment by seisman** [comment]
Please also post the output of the `gs --help` command.

**Comment by jsbyrnes** [comment]
(pygmt) ***@***.*** ~ % which gs

/Users/josephbyrnes/opt/anaconda3/envs/pygmt/bin/gs





and





(pygmt) ***@***.*** ~ % gs --help

GPL Ghostscript 10.02.1 (2023-11-01)

Copyright (C) 2023 Artifex Software, Inc.  All rights reserved.

Usage: gs [switches] [file1.ps file2.ps ...]

Most frequently used switches: (you can use # in place of =)

 -dNOPAUSE           no pause after page   | -q       `quiet', fewer
messages

 -g<width>x<height>  page size in pixels   | -r<res>  pixels/inch resolution

 -sDEVICE=<devname>  select device         | -dBATCH  exit after last file

 -sOutputFile=<file> select output file: - for stdout, |command for pipe,

                                         embed %d or %ld for page #

Input formats: PostScript PostScriptLevel1 PostScriptLevel2
PostScriptLevel3 PDF

Default output device: bbox

Available devices:

   alc1900 alc2000 alc4000 alc4100 alc8500 alc8600 alc9100 ap3250 appledmp

   atx23 atx24 atx38 bbox bit bitcmyk bitrgb bitrgbtags bj10e bj10v bj10vh

   bj200 bjc600 bjc800 bjc880j bjccmyk bjccolor bjcgray bjcmono bmp16 bmp16m

   bmp256 bmp32b bmpgray bmpmono bmpsep1 bmpsep8 ccr cdeskjet cdj1600 cdj500

   cdj550 cdj670 cdj850 cdj880 cdj890 cdj970 cdjcolor cdjmono cdnj500 cfax

   chp2200 cif cljet5 cljet5c cljet5pr coslw2p coslwxl declj250 deskjet

   devicen dfaxhigh dfaxlow display dj505j djet500 djet500c dl2100 dnj650c

   docxwrite epl2050 epl2050p epl2120 epl2500 epl2750 epl5800 epl5900

   epl6100 epl6200 eplcolor eplmono eps2write eps9high eps9mid epson epsonc

   escp escpage faxg3 faxg32d faxg4 fmlbp fmpr fpng fs600 gdi hl1240 hl1250

   hl7x0 hpdj1120c hpdj310 hpdj320 hpdj340 hpdj400 hpdj500 hpdj500c hpdj510

   hpdj520 hpdj540 hpdj550c hpdj560c hpdj600 hpdj660c hpdj670c hpdj680c

   hpdj690c hpdj850c hpdj855c hpdj870c hpdj890c hpdjplus hpdjportable ibmpro

   ijs imagen inferno ink_cov inkcov itk24i itk38 iwhi iwlo iwlq jetp3852

   jj100 jpeg jpegcmyk jpeggray la50 la70 la75 la75plus laserjet lbp310

   lbp320 lbp8 lex2050 lex3200 lex5700 lex7000 lips2p lips3 lips4 lips4v

   lj250 lj3100sw lj4dith lj4dithp lj5gray lj5mono ljet2p ljet3 ljet3d ljet4

   ljet4d ljet4pjl ljetplus ln03 lp1800 lp1900 lp2000 lp2200 lp2400 lp2500

   lp2563 lp3000c lp7500 lp7700 lp7900 lp8000 lp8000c lp8100 lp8200c lp8300c

   lp8300f lp8400f lp8500c lp8600 lp8600f lp8700 lp8800c lp8900 lp9000b

   lp9000c lp9100 lp9200b lp9200c lp9300 lp9400 lp9500c lp9600 lp9600s

   lp9800c lps4500 lps6500 lq850 lxm3200 lxm5700m m8510 md1xMono md2k

   md50Eco md50Mono md5k mgr4 mgr8 mgrgray2 mgrgray4 mgrgray8 mgrmono miff24

   mj500c mj6000c mj700v2c mj8000c ml600 necp6 npdl nullpage oce9050 oki182

   oki4w okiibm oprp opvp paintjet pam pamcmyk32 pamcmyk4 pbm pbmraw pcl3

   pclm pclm8 pcx16 pcx24b pcx256 pcxcmyk pcxgray pcxmono pdfimage24

   pdfimage32 pdfimage8 pdfwrite pgm pgmraw pgnm pgnmraw photoex picty180 pj

   pjetxl pjxl pjxl300 pkm pkmraw pksm pksmraw plan plan9bm planc plang

   plank planm plib plibc plibg plibk plibm png16 png16m png16malpha png256

   png48 pngalpha pnggray pngmono pngmonod pnm pnmraw ppm ppmraw pr1000

   pr1000_4 pr150 pr201 ps2write psdcmyk psdcmyk16 psdcmykog psdcmyktags

   psdcmyktags16 psdrgb psdrgb16 psdrgbtags pxlcolor pxlmono r4081 rinkj

   rpdl samsunggdi sj48 spotcmyk st800 stcolor t4693d2 t4693d4 t4693d8

   tek4696 tiff12nc tiff24nc tiff32nc tiff48nc tiff64nc tiffcrle tiffg3

   tiffg32d tiffg4 tiffgray tifflzw tiffpack tiffscaled tiffscaled24

   tiffscaled32 tiffscaled4 tiffscaled8 tiffsep tiffsep1 txtwrite uniprint

   xcf xes xpswrite

Search path:

   /Applications/
GMT-6.4.0.app/Contents/Resources/share/ghostscript/9.53.3/Resource/Init :

   %rom%Resource/Init/ : %rom%lib/ :

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/ghostscript/10.02.1/Resource/Init
:

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/ghostscript/10.02.1/lib
:

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/ghostscript/10.02.1/Resource/Font
:

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/ghostscript/fonts :

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/fonts/default/ghostscript
:

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/fonts/default/Type1 :

   /Users/josephbyrnes/opt/anaconda3/envs/pygmt/share/fonts/default/TrueType
:

   /usr/lib/DPS/outline/base : /usr/openwin/lib/X11/fonts/Type1 :

   /usr/openwin/lib/X11/fonts/TrueType

Initialization files are compiled into the executable.

For more information, see
https://ghostscript.readthedocs.io/en/gs10.02.1/Use.html

Please report bugs to bugs.ghostscript.com.

**Comment by seisman** [comment]
From the output, I can see that you have two gs versions installed:

- 10.02.1 installed via conda `/Users/josephbyrnes/opt/anaconda3/envs/pygmt/bin/gs`
- 9.53.3 shipped with the GMT 6.4.0 app installer `   /Applications/
GMT-6.4.0.app/Contents/Resources/share/ghostscript/9.53.3/Resource/Init:`

It's very likely your gs 10.02.1 is reading the gs resources from the one provided by GMT. So the easiest solution is to fully remove the GMT-6.4.0 app from your system.

**Comment by jsbyrnes** [comment]
This solved this issue like a charm - in case anyone else has a
similar issue, the problem was with my .zshrc file - even with GMT-6.4.0
technically uninstalled, its paths were being added when I opened the
terminal. Removing those fixed the default installation from the pygmt
site.

Thank you for the prompt help.
